src/runtime_src/core/edge/user/device_linux.cpp

```diff
@@ -42,6 +42,47 @@ get_edgedev(const xrt_core::device* device)
   return zynq_device::get_dev();
 }
 
+struct bdf 
+{
+  using result_type = query::pcie_bdf::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type)
+  {
+    return std::make_tuple(0,0,0);
+  }
+
+};
+
+struct board_name 
+{
+  using result_type = query::board_name::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type)
+  {
+    result_type deviceName("edge");
+    std::ifstream VBNV;
+    VBNV.open("/etc/xocl.txt");
+    if (VBNV.is_open()) {
+      VBNV >> deviceName;
+    }
+    VBNV.close();
+    return deviceName;
+  }
+};
+
+struct is_ready 
+{
+  using result_type = query::is_ready::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type)
```

Comment:
Why pointer and not reference?

Reply:
@keryell  This is called from a shared_ptr.get().  It is a normal and good idiom to extract pointer from the smart pointer when calling a function where arguments stay in scope for the duration of the call.   It is extremely awkward in my opinion to deference the pointer to convert it to a reference.

Reply:
I see. I am just browsing the PR without the big picture... :-/

Reply:
Understood :-)